storage/remote/queue_manager.go

```diff
@@ -261,6 +264,20 @@ type QueueManager struct {
 	samplesIn, samplesDropped, samplesOut, samplesOutDuration *ewmaRate
 
 	metrics *queueManagerMetrics
+
+	endpointRecord EndpointRecord
+}
+
+// EndpointRecord Structure holds the segment the endpoint url
+type EndpointRecord struct {
+	Segment  string
```

Comment:
the fields on this and checkpoint record don't need to be exported if you don't have a need to access those fields directly outside of the `storage/remote` package.